### PR TITLE
Build working Meshroom images based on the latest Alicevision images

### PR DESCRIPTION
**Pull request by deece**
## Description

This patch builds upon the AliceVision PR:
https://github.com/alicevision/AliceVision/pull/853

It ensures that Centos images build a functioning Meshroom,
and also adds Ubuntu support.

The Docker images support launch the GUI via ssh:
docker run -it --runtime nvidia -p 2222:22 --name meshroom -v</path/to/your/data>:/data alicevision/meshroom:<meshroom branch/commit>-av<AliceVision branch/commit>-ubuntu20.04-cuda11.0
ssh -p 2222 -X root@<docker host> /opt/Meshroom_bundle/Meshroom # Password is 'meshroom'

The following OS/CUDA versions are supported:
CUDA_VERSION=11.0 UBUNTU_VERSION=20.04
CUDA_VERSION=11.0 UBUNTU_VERSION=18.04
CUDA_VERSION=10.2 UBUNTU_VERSION=18.04
CUDA_VERSION=9.2 UBUNTU_VERSION=18.04

CUDA_VERSION=10.2 CENTOS_VERSION=7
CUDA_VERSION=9.2 CENTOS_VERSION=7

To build all supported images:
AV_VERSION=<AliceVision version> ./docker-build-all.sh

## Features list
- [X] Build working images for Ubuntu 20.04
- [X] Build working images for Ubuntu 18.04
- [X] Build working images for CentOS 7
- [X] Enable remote GUI launch via SSH

### CODE REVIEW

**Comment by fabiencastan** [comment]
Hi @deece ,
Thanks for the very nice contribution!
I have put some comments on the AliceVision PR that needs to be merged first.

We start to have many files related to docker in the root folder.
It would be good to find a way to move them into the docker folder.

**Comment by fabiencastan** [comment]
@deece I saw that you have moved all docker related files into a "docker" folder on the AliceVision side. Could you do the same here?

File: docker-build.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+test -z "$MESHROOM_VERSION" && MESHROOM_VERSION="$(git rev-parse --abbrev-ref HEAD)-$(git rev-parse --short HEAD)"
+test -z "$AV_VERSION" && echo "AliceVision version not specified, set AV_VERSION in the environment" && exit 1
+test -z "$CUDA_VERSION" && CUDA_VERSION="10.2"
+test -z "$CENTOS_VERSION" && CENTOS_VERSION=7
+
+test -d dl || \
+        mkdir dl
+test -f dl/qt.run || \
+        wget "https://download.qt.io/archive/qt/5.14/5.14.1/qt-opensource-linux-x64-${QT_VERSION_B}.run" -O "dl/qt.run"
```
Comment (fabiencastan) [inline docker-build.sh:12]:
```suggestion
        wget "https://download.qt.io/archive/qt/5.14/5.14.1/qt-opensource-linux-x64-5.14.1.run" -O "dl/qt.run"
```

**Comment by deece** [comment]
I'm moving the scripts & Dockerfiles into the docker subdir atm. I haven't updated Dockerfile_py2, as Python2 is EOL. Should I remove it instead?

**Comment by fabiencastan** [comment]
I still need Python 2 for production for now.

File: Dockerfile [hunk 2]
```diff
@@ -1,39 +0,0 @@
...
-
-# Build Meshroom plugins
-RUN cmake "${MESHROOM_DEV}" -DALICEVISION_ROOT="${AV_INSTALL}" -DQT_DIR="${QT_DIR}" -DCMAKE_INSTALL_PREFIX="${MESHROOM_BUNDLE}/qtPlugins"
-# RUN make -j8 qtOIIO
-# RUN make -j8 qmlAlembic
-# RUN make -j8 qtAliceVision
-RUN make -j8 && cd /tmp && rm -rf ${MESHROOM_BUILD}
-
-RUN mv "${AV_BUNDLE}" "${MESHROOM_BUNDLE}/aliceVision"
-RUN rm -rf ${MESHROOM_BUNDLE}/aliceVision/share/doc ${MESHROOM_BUNDLE}/aliceVision/share/eigen3 ${MESHROOM_BUNDLE}/aliceVision/share/fonts ${MESHROOM_BUNDLE}/aliceVision/share/lemon ${MESHROOM_BUNDLE}/aliceVision/share/libraw ${MESHROOM_BUNDLE}/aliceVision/share/man/ aliceVision/share/pkgconfig
```
Comment (fabiencastan) [inline Dockerfile:37]: This is needed

**Comment by fabiencastan** [comment]
Merged by another PR https://github.com/alicevision/meshroom/pull/1054.

Thanks for the great work. Do not hesitate to create another PR if you want to make adjustments to the changes that I have made to integrate it.

**Pull request closed without merging**